samples/asgardeo-react-app/src/error-boundary.tsx

```diff
@@ -0,0 +1,18 @@
+import { AsgardeoAuthException } from "@asgardeo/auth-react";
+import React, { FunctionComponent, ReactElement } from "react";
+import { InvalidSystemTimePage } from "./pages/InvalidSystemTime";
+
+interface ErrorBoundaryProps {
+  error: AsgardeoAuthException;
+  children: ReactElement;
+}
+
+export const ErrorBoundary: FunctionComponent<ErrorBoundaryProps> = (
+  props: ErrorBoundaryProps
+): ReactElement => {
+  const { error, children } = props;
+
+  return error?.code === "JS-CRYPTO_UTILS-IVIT-IV02" ? <InvalidSystemTimePage /> : error ? null : (
```

Comment:
The id token validation may fail for different reasons. We can't say for sure if it is because of invalid system time.

Reply:
This improvement will be tracked from https://github.com/wso2-enterprise/asgardeo-product/issues/10296